.github/workflows/push.yml

```diff
@@ -23,23 +23,65 @@ jobs:
         cd docs
         make html-author-mode
         grep -r "page unresolved" target/staging/docs/ && exit 1 || echo 'ok'
-  build-and-test-scala:
+
+  build-and-test-core:
+    name: "Cloudflow Core"
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        java: [ '8', '11', '13' ]
     steps:
-    - uses: actions/checkout@v2
 
-    - name: Setup environment
-      run: |
-        make -C .github/workflows/setup setup-local-all
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
 
     - name: Cache SBT and Coursier cache
       uses: coursier/cache-action@v3
 
+    - name: Scalafmt check
+      run: cd core && sbt "scalafmtCheckAll; scalafmtSbtCheck"
+
     - name: build-and-test
       env:
         SBT_OPTS: -Xms512M -Xmx2048M -Xss2M -XX:MaxMetaspaceSize=1024M
-      run: |
-        export PATH="$HOME/opt/sbt/bin:$PATH"
-        export SBT_OPTS="-Xss256M"
-        cd scripts
-        ./build-all.sh +test
+      run: ./scripts/build-core.sh +test
+
+  build-and-test-sbt-examples:
+    name: "sbt samples"
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        java: [ '8', '11', '13' ]
```

Comment:
Perhaps we don't want these samples to run on all these JDK versions.

Reply:
agreed samples can run on java 8 only